src/ua_types_encoding_binary.c

```diff
@@ -708,13 +728,19 @@ ExtensionObject_encodeBinary(UA_ExtensionObject const *src, const UA_DataType *_
         UA_NodeId typeId = src->content.decoded.type->typeId;
         if(typeId.identifierType != UA_NODEIDTYPE_NUMERIC)
             return UA_STATUSCODE_BADENCODINGERROR;
-        typeId.identifier.numeric += UA_ENCODINGOFFSET_BINARY;
+        const UA_DataType *type = NULL;
+        /* helping clang analyzer, typeId is numeric */
+        UA_assert(typeId.identifier.byteString.data == NULL);
+        UA_assert(typeId.identifier.string.data == NULL);
+        if (findDataType(&typeId, &type)!= UA_STATUSCODE_GOOD)
```

Comment:
We should not do a findDataType here.
We know the type. It is `src->content.decoded.type`.

Otherwise, ready to merge!